web/src/CityContentSwitcher.tsx

```diff
@@ -127,9 +123,13 @@ const CityContentSwitcher = ({ cities, languageCode }: CityContentSwitcherProps)
       element={
         <Suspense
           fallback={
-            <CityContentLayout {...suspenseLayoutProps} route={route}>
-              <LoadingSpinner />
-            </CityContentLayout>
+            city ? (
+              <CityContentLayout {...suspenseLayoutProps} route={route} city={city}>
+                <LoadingSpinner />
+              </CityContentLayout>
+            ) : (
+              <Layout />
+            )
           }>
           <Component {...cityRouteProps} />
```

Comment:
so we could also change the `CityRouteProps` to 
```ts
type CityRouteProps {
  city: CityModel
  pathname: string
  cityCode: string
  languageCode: string
}
```
and then we can write sth like: 
```suggestion
          city ? <Component city={city} pathname={pathname} cityCode={cityCode} languageCode={languageCode} /> : null
```
and get rid of all the `if (!city)` in the routes right?

Reply:
Nope, actually we can't, thats part of the whole PR. By rendering the componentns/routes even if the city is not available, we trigger the loading of the routes data (i.e. events, categories, ...). If we only render the component if the city is available, it will be subsequent requests, not simultaneous ones.